### ISSUE TITLE
Invalid memory access error related to blocks and closures

### ISSUE THREAD

**Issue by dawkot** [crystal-lang/crystal#6693]
```crystal
def m(&f)
  f :s
end

def m(f)
  m f
end

m ->(x : Symbol) { x }
```

> Invalid memory access (signal 11) at address 0x7ffc2b57dff8
> [0x557c806deab6] *CallStack::print_backtrace:Int32 +118
> [0x557c806d3a1d] __crystal_sigfault_handler +61
> [0x7f6814a913c0] ???
> [0x557c806d3a61] *m<Proc(Symbol, Symbol)>:NoReturn +1
> [0x557c806d3a66] ??? (524028 times)
> [0x557c806c6abe] __crystal_main +1550
> [0x557c8071b026] *Crystal::main_user_code<Int32, Pointer(Pointer(UInt8))>:Nil +6
> [0x557c8071af89] *Crystal::main<Int32, Pointer(Pointer(UInt8))>:Int32 +41
> [0x557c806d1a66] main +6
> [0x7f681465e223] __libc_start_main +243
> [0x557c806c63de] _start +46
> [0x0] ???

**Comment by asterite** [comment]
```crystal
def m(f)
  m f
end
```

This is an infinite recursive call. You can tell by the (kind of broken) stack trace:

```
[0x557c806d3a66] ??? (524028 times)
```

**Comment by asterite** [comment]
Closed as duplicate of #271
